index.bs

```diff
@@ -641,7 +643,97 @@ To <dfn>obtain a report</dfn> given an [=attribution source=] |source| and an [=
+     closed. Adding random delay prevents temporal joining of reports from different [=attribution source/source origin=]'s.
+1. Wait until |report|'s [=attribution report/report time=] is the current time.
+1. Optionally, wait a further [=implementation-defined=] length of time.
+    
+    Note: This is intended to allow user agents to optimize device resource usage.
+1. Run [=attempt to deliver a report=] with |report|.
+
+<h3 id="serialize-report-body">Serialize attribution report body</h3>
+
+To <dfn>serialize an [=attribution report=]</dfn> |report|, run the following steps:
+
+1. Let |data| be a [=map=] of the following key/value pairs:
+
+    : `"source_type"`
+    :: |report|'s [=attribution report/source type=]
+    : `"source_event_id"`
+    :: |report|'s [=attribution report/event id=]
+    : `"trigger_data"`
+    :: |report|'s [=attribution report/trigger data=]
+
+1. Return the [=byte sequence=] resulting from executing [=serialize an infra value to JSON bytes=] on |data|.
+
+<h3 id="get-report-url">Get report request URL</h3>
+
+To <dfn>generate a report URL</dfn> for a given [=attribution report=] |report|, run the following steps:
+
+1. Let |reportUrl| be a new [=URL record=].
+1. Let |reportingOrigin| be |report|'s [=attribution report/reporting endpoint=].
+1. Set |reportUrl|'s [=URL/scheme=] to |reportingOrigin|'s [=origin/scheme=].
+1. Set |reportUrl|'s [=URL/host=] to |reportingOrigin|'s [=origin/host=].
+1. Set |reportUrl|'s [=URL/port=] to |reportingOrigin|'s [=origin/port=].
+1. Set |reportUrl|'s [=URL/path=] to «"`.well-known`","`attribution-reporting`","`report-attribution`"».
+1. Return |reportUrl|.
+
+<h3 id="issue-report-request">Issuing a report request</h3>
+
+This algorithm constructs a [=request=] and attempts to deliver it to
+|report|'s [=attribution report/reporting endpoint=].
+
+To <dfn>attempt to deliver a report</dfn> given an [=attribution report=] |report| run the following steps:
+
+1. Let |body| be the result of executing [=serialize an attribution report=] on |report|.
+1. Let |url| be the result of executing [=generate a report URL=] on |report|.
+1. Let |referrer| the result of running the [=URL parser=] on |report|'s [=attribution report/attribution destination=].
+1. Let |request| be a new [=request=] with the following properties:
+    :   `method`
+    ::  "`POST`"
+    :   `url`
+    ::  |url|
+    :   `header list`
+    ::  A new <a>header list</a> containing a <a>header</a> named
+        "`Content-Type`" whose value is "`application/json`"
+    :   `body`
+    ::  A [=/body=] whose [=body/source=] is |body|.
+    : `referrer`
+    :: |referrer|
+    :   `client`
+    ::  `null`
+    :   `window`
+    ::  "`no-window`"
+    :   `service-workers mode`
+    ::  "`none`"
+    :   `initiator`
+    ::  ""
+    :   `mode`
+    ::  "`cors`"
+    :   `unsafe-request` flag
+    ::  set
+    :   `credentials mode`
+    ::  "`omit`"
+    :   `cache mode`
+    ::  "`no-store`"
```

Comment:
Should these all nest the double-quotes within the backticks instead of vice versa?

Reply:
I believe "<code></code>" is the standard (at least in the infra/fetch specs). Updated throughout for consistency.